Sources/ImgixSwift/ImgixClient.swift

```diff
@@ -98,6 +98,7 @@ import Foundation
         }
         else {
             path = path.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed) ?? path
+            path = path.replacingOccurrences(of: "+", with: "%2B")
```

Comment:
Instead of `replacingOccurrences` maybe we could drop `+` and `:` from the `urlPathAllowed` character set?

Also, I'm wondering how `:` is getting encoded... Does this test actually pass for you?

```swift
    func testEncodesReservedCharacters() {
         let generatedUrl = client.buildUrl("&$+,:;=?@#.jpg", params: [:])
         let expectedUrl = "https://paulstraw.imgix.net/&$%2B,%3A%3B=%3F@%23.jpg"

         XCTAssert(generatedUrl.absoluteString == expectedUrl)
     }
```